redash/models/__init__.py

```diff
@@ -64,6 +65,58 @@ def get(self, query_id):
 scheduled_queries_executions = ScheduledQueriesExecutions()
 
 
+@python_2_unicode_compatible
+class TableMetadata(TimestampMixin, db.Model):
+    id = Column(db.Integer, primary_key=True)
+    data_source_id = Column(db.Integer, db.ForeignKey("data_sources.id"))
+    table_exists = Column(db.Boolean, default=True)
```

Comment:
We store a boolean of whether the object exists, but we need some indication on when we can delete them. We could use `updated_at` as the time it was detected as missing, but for this to work we need to make sure the records are not touched after being marked as exists=False.

Reply:
I made this adjustment. Now records set to exists=False do not get updated and they set the `updated_at` timestamp which is periodically checked in a celery task to prune old data